jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -104,13 +104,13 @@ public ArrayByteBufferPool(int minCapacity, int factor, int maxCapacity, int max
         _minCapacity = minCapacity;
 
         // Initialize all buckets in constructor and never modify the array again.
-        int length = bucketFor(maxCapacity);
+        int length = bucketFor(maxCapacity) + 1;
```

Comment:
what is the extra allocated bucket used for?

Reply:
@gregw `bucketFor(capacity)` gives you the index of the bucket of that capacity. So the size of the array is `bucketFor(maxCapacity) + 1`. The bucket at index 0 now contains the buffers of capacity 0 and I wasn't including this before.

Also the test failure was unrelated, the log said `Error cloning remote repo 'origin'`, so I have started a re-run of the tests.